exercises/concept/animal-magic/animal_magic_test.go

```diff
@@ -55,11 +58,22 @@ func TestWandEnergy(t *testing.T) {
 		if i > 0 && got != last {
 			foundDifferent = true
 		}
+		buckets[int(got/bucketSize)]++
 		last = got
 	}
 	if !foundDifferent {
 		t.Errorf("GenerateWandEnergy() always generates the same number: %f", got)
 	}
+
+	var emptyBuckets int
+	for _, v := range buckets {
+		if v == 0 {
+			emptyBuckets++
+		}
+	}
+	if emptyBuckets > 1 {
+		t.Errorf("GenerateWandEnergy() results are not uniformly distributed. %d of %d buckets are empty.", emptyBuckets, numBuckets)
```

Comment:
For concept exercises, the student cannot see the test file when working on the website. That means it is important that the error messages are understandable without seeing the code.

I would recommend not talk about "buckets", instead either write a general statement that some sections of the range are not covered by the results or and output one example of an empty section (which you would need to "remember" from the loop above).

Reply:
Makes sense. Also no reason to run the entire loop once a failure is detected.